api.yaml

```diff
@@ -126,6 +182,51 @@ components:
                        Binary caches are checked for binaries in order of ascending priority;
                        thus a higher number denotes a lower priority.
                        The binary cache https://cache.nixos.org has priority 40.
+    FileListingEntryType:
+      type: string
+      enum: [directory, regular]
+    FileListingDirectoryEntry:
+      type: object
+      required:
+        - type
+        - entries
+      properties:
+        type:
+          $ref: '#/components/schemas/FileListingEntryType'
+        entries:
+          type: object
+          additionalProperties:
+            oneOf:
+              - $ref: '#/components/schemas/FileListingFileEntry'
+              - $ref: '#/components/schemas/FileListingDirectoryEntry'
+    FileListingFileEntry:
+      type: object
+      required:
+        - type
+        - size
+        - narOffset
+      properties:
+        type:
+          $ref: '#/components/schemas/FileListingEntryType'
+        size:
+          type: integer
```

Comment:
Honestly not sure the best way to do this with the OpenAPI spec.
I could consider just merging it into a single type...